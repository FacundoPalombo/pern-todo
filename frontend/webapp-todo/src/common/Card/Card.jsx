import React from "react";
import styles from "./Card.module.scss";

export const Card = ({ children }) => {
  return <div className={styles.container}>{children}</div>;
};
