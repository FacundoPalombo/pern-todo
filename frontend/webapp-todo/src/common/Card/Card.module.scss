.container {
  padding: 20px;
  margin: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 300px;
  height: auto;
}
