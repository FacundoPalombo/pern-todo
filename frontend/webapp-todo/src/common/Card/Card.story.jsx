import React from "react";
import { Card } from "./Card";

export const withChildren = () => (
  <Card>
    <div style={{ width: "200px", height: "200px" }}>
      <div
        style={{ background: "rebeccapurple", width: "95%", height: "95%" }}
      />
    </div>
  </Card>
);
export const withoutChildren = () => <Card />;

export default {
  title: "Card",
  component: Card,
};
