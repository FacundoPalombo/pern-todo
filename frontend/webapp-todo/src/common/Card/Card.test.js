import React from "react";
import { render } from "@testing-library/react";
import { Card } from "./Card";

describe("GIVEN Card common component", () => {
  const { asFragment } = render(<Card />);
  const firstRender = asFragment();

  it("THEN should render", () => {
    expect(firstRender).toMatchSnapshot();
  });
  it("THEN should render with children", () => {
    const { asFragment } = render(
      <Card>
        <div id="testSelector"></div>
      </Card>
    );
    expect(asFragment().querySelector("#testSelector")).not.toBeNull();
  });
});
