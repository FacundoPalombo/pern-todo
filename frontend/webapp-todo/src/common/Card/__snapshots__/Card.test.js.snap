// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Card common component THEN should render 1`] = `
<DocumentFragment>
  <div
    class="container"
  />
</DocumentFragment>
`;
